into/backends/csv.py

```diff
@@ -277,4 +278,13 @@ def infer_header(csv, encoding='utf-8', **kwargs):
             not all(dt == 'O' for dt in df.dtypes))
 
 
+def get_delimiter(csv, length=8192):
+    """Figure out the delimiter of a CSV file.
+    """
+    sniffer = Sniffer()
+    with sample(csv, length=length) as fn:
+        with open(fn, mode='rb') as f:
+            return sniffer.sniff(f.read()).delimiter
```

Comment:
Suggest generalizing to `get_dialect` which returns a dict with all of the delimiter terms